src/main/java/org/fenixedu/academic/domain/studentCurriculum/StudentCurricularPlanEnrolment.java

```diff
@@ -380,31 +381,45 @@ protected boolean isResponsiblePersonCoordinator() {
     // -------------------
 
     static public StudentCurricularPlanEnrolment createManager(final EnrolmentContext enrolmentContext) {
+        return ENROLMENT_MANAGER_FACTORY.get().createManager(enrolmentContext);
+    }
 
-        if (enrolmentContext.isNormal()) {
+    static public Supplier<EnrolmentManagerFactory> ENROLMENT_MANAGER_FACTORY = () -> new EnrolmentManagerFactory() {
```

Comment:
Don't publicly expose an internal field, keep it `private` and expose a setter (which checks that the new factory is not `null`).